### ISSUE TITLE
Fade example in documentation doesn't work

### ISSUE THREAD

**Issue by CGamesPlay** [reactjs/react-transition-group#156]
The example in the documentation for CSSTransition doesn't work.

**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**

https://jsfiddle.net/m4mb9Lg3/13/

Notice that the text fades out but then immediately reappears, then a few seconds later immediately disappears and fades back in. There doesn't seem to be any code for applying a className while entered after transitions have finished.

**What is the expected behavior?**

The words "Hello World" should fade in and out.

**Which versions, and which browser / OS are affected by this issue? Did this work in previous versions?**

I'm running Chrome Version 60.0.3112.90 (Official Build) (64-bit) and have never seen this in previous versions.

**Comment by AlesMenzel** [comment]
**Can confirm.** Actually trying to implement it and the elements are reappearing instantly after the exit classes are removed - which makes sence, but the documentation does not specify if we should manually remove the nodes after the set timeout ?

**Comment by AlesMenzel** [comment]
It seems that the correct way of implementing CSSTransition is as follows:

```jsx
<TransitionGroup>
	{props.display and
		<CSSTransition key={0} timeout={5000} classNames="loader__fade">
			<Loader />
		</CSSTransition>
	}
</TransitionGroup>
```

even for a single child.

**Comment by jquense** [comment]
You don't need transition group for single children.  The fiddle in the issue template demonstrates a fade. If you want to remove an element when it exists use the unmountOnExit prop or tweak the styles to work with an element that stays mounted. I'll double check the example css

**Comment by AlesMenzel** [comment]
You are right, i have updated @CGamesPlay example: https://jsfiddle.net/m4mb9Lg3/19/
The documentation regarding this is little confusing though.

**Comment by CGamesPlay** [comment]
Let me enumerate some configurations:

|unmountOnExit|state|notes|
|--|--|--|
|false|ENTERING|className has enter and enterActive appended|
|false|ENTERED|className is unmodified|
|false|EXITING|className has exit and exitActive appended|
|false|EXITED|className is unmodified|
|true|ENTERING|className has enter and enterActive appended|
|true|ENTERED|className is unmodified|
|true|EXITING|className has exit and exitActive appended|
|true|EXITED|node does not exist|

The problem I'm seeing here is that `(false, ENTERED)` and `(false, EXITED)` are identical so in order to successfully use `unmountOnExit={false}` (the default) I have to build a stateful component to track whether state is `ENTERED` or `EXITED`.

This problem would be sidestepped if the exit and exitActive classes were kept on the exited node. It looks like `unmountOnExit` is the target use case for this library, so `unmountOnExit={false}` probably wasn't fully thought through when writing this module.

**Comment by jquense** [comment]
Yeah we may add some exited/entered classes. You are correct though, what happened is that the v1 api was tied to the TransitionGroup group use case which always unmounts on exit so the trailing classes weren't needed. We tried to keep the API very close for ease of upgrading but your right it's not super helpful as a standalone component. I hope we can change that! 

In the meantime maybe check out the note in the docs about the react-bootstrap implementations, they do what you want but don't use CssTransition

**Comment by pashaproton** [comment]
In the example Transition works for already created element. But what about dynamically added elements? I cannot figure out how to make possible animate inserted element. Maybe someone has example? Even example from documentation doesn't work...
